viral/sequence_utils.py

```
@@ -0,0 +1,924 @@
+    """
+    "Where posj is the jth spatial bin, bini is the ith temporal (two frame) bin in the event,
+    Prij is the Bayesian posterior probability for that spatial bin at that temporal bin,
+    M is the total number of temporal bins and N is the total number of spatial bins."
+
+    Essentially, this is a Python implementation of https://github.com/losonczylab/Grosmark_NatNeuro_2021/blob/main/calcWeightedLinearCorr.m
+    """
+    w = posterior_probability_matrix.flatten(
+        order="F"
+    )  # column-major flattening like matlab
+    if np.isnan(w).any():
+        raise ValueError("Encountered NaN in posterior probability matrix")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)